src/HttpHandler.php

```diff
@@ -12,41 +12,46 @@
 
 class HttpHandler implements RequestHandlerInterface
 {
-    private RouteDispatcher $dispatcher;
+    private HandlerConfig $routeConfig;
     private ResponseFactoryInterface $responseFactory;
 
-    public function __construct(RouteDispatcher $routerDispatcher, ResponseFactoryInterface $responseFactory)
+    public function __construct(HandlerConfig $routeConfig, ResponseFactoryInterface $responseFactory)
     {
-        $this->dispatcher = $routerDispatcher;
+        $this->routeConfig = $routeConfig;
         $this->responseFactory = $responseFactory;
     }
 
     public function handle(ServerRequestInterface $request): ResponseInterface
     {
-        $routeInfo = $this->dispatcher->dispatch($request);
+        $info = $this->routeConfig->getHandlerInfo($request);
 
-        if ($routeInfo->notAllowed()) {
+        if ($info->handlerNotAllowed()) {
             return $this->responseFactory->createResponse(405, 'Method Not Allowed');
         }
 
-        if ($routeInfo->notFound()) {
+        if ($info->handlerNotFound()) {
             return $this->responseFactory->createResponse(404, 'Not Found');
         }
 
-        return $this->executeHandler($routeInfo->getHandler(), $routeInfo->getArguments());
+        try {
+            return $this->executeRouteHandler($info->getHandler(), $info->getArguments());
+        } catch (Throwable $e) {
+            return $this->handleException();
+        }
     }
 
     /**
      * @param mixed[] $arguments
      */
-    private function executeHandler(callable $handler, array $arguments): ResponseInterface
+    private function executeRouteHandler(callable $handler, array $arguments): ResponseInterface
     {
-        try {
-            call_user_func_array($handler, $arguments);
-        } catch (Throwable $e) {
-            return $this->responseFactory->createResponse(500, 'Internal Server Error');
-        }
+        call_user_func_array($handler, $arguments);
 
         return $this->responseFactory->createResponse(200, 'OK');
     }
+
+    private function handleException(): ResponseInterface
+    {
+        return $this->responseFactory->createResponse(500, 'Internal Server Error');
```

Comment:
Just a comment for the future implementation: there might be no need to have exception handling in this package. I mean, the user will be capable to set up a hashmap with callbacks for handling exception (as you suggested). This callback can receive the exception and a response. However, if we want to implement something like `HttpException`, it could be in a completely separated package defining these `HttpException` classes and a `HttpExceptionCatcherMiddleware`. What do you think?